More advanced elaboration. (#3241)
modelica/ModelicaSpecification: chapters/connectors.tex

This applies to both variables of simple types, and variables of structured types.

Then connections containing expandable connectors are elaborated:
\begin{enumerate}
\item
  If a connect-statement references a potentially present component (either as the complete argument or as part of the argument) it will be marked as being present and will be treated as declared below.
\item
  After that at least one connector in the connect equation must reference a declared component.
\item
  If the other connector is undeclared it must be in a declared component and is handled as follows:

> \begin{itemize}

Reviewer: Match indentation of items as well as `\end` below:
```suggestion
  \begin{itemize}
```

This applies to both variables of simple types, and variables of structured types.

Then connections containing expandable connectors are elaborated:
\begin{enumerate}
\item

>   If a connect-equation references a potentially present component (either as the complete argument or as part of the argument) it will be marked as being present and will be treated below as declared.

Author: ```suggestion
  If a connect-equation references a potentially present component \emph{as part of the argument} it will be marked as being present, which will allowing a connection to an undeclared connector inside it.
  The rule does not apply for the complete argument.
```
As discussed.

>   If a connect-statement references a potentially present component (either as the complete argument or as part of the argument) it will be marked as being present and will be treated as declared below.

Reviewer 2: My first interpretation of "will be treated as declared below" was that it would be declared somewhere below. Perhaps changing the order would avoid this confusion:
```suggestion
  If a connect-statement references a potentially present component (either as the complete argument or as part of the argument) it will be marked as being present and will be treated below as declared.
```

Author: Ok, seems good.

This applies to both variables of simple types, and variables of structured types.

Then connections containing expandable connectors are elaborated:
\begin{enumerate}
\item

>   If a connect-equation references a potentially present component \emph{as part of the argument} it will be marked as being present, which will allowing a connection to an undeclared connector inside it.

Reviewer 2: Typo:
```suggestion
  If a connect-equation references a potentially present component \emph{as part of the argument} it will be marked as being present, which will allow a connection to an undeclared connector inside it.
```

    \end{nonnormative}
    For an array the \lstinline!input!/\lstinline!output! property can be deduced separately for each array element.

>   \end{itemize}

Reviewer: Match indentation of corresponding `\begin`:
```suggestion
\end{itemize}
```
